tests/jballerina-integration-test/src/test/resources/http/src/httpservices/47_cache_validation_request_header_preservation.bal

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+http:Client cachingEP4 = new("http://localhost:9260", { cache: { isShared: true } });
```

Comment:
With this refactoring shall we structure the test services into few listeners? Because all of them have common configurations and each consists of only a single resource. I think it's possible that we can have two listeners, one for passthrough service which does the client call, and one for the backend. Use a descriptive resource path instead of basePath.

Following files can be refactored : 
37_no_cache_service.bal
38_max_age_cache_service.bal
39_must_revalidate_cache_service.bal
47_cache_validation_request_header_preservation.bal